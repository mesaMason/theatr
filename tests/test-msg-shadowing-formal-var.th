dolphin(int weight, int age):
    print("inside dolphin.")
    print("dolphins local sum var =")
    int sum = weight + age
    print(sum)

    receive:
        // msg local var shadowing msg formal var
        swim(int hours):
            int hours = 0
            print("inside swim().")
            print("dolphins hours =")
            print(hours)

        // msg formal shadowing actor formal
        eat(int weight):
            print("inside eat().")
            print("eat()s formal weight =")
            print(weight)

        // msg formal shadowing actor local
        sleep(int sum):
            print("inside sleep().")
            print("sleep()s local sum =")
            print(sum)
    drop:
        print("inside drop().")
        print("dolphins formal weight = ")
        print(weight)
        print("dolphins local sum = ")
        print(sum)
    after:
        return;
        
func main() -> int:
    int weight = 10
    int age = 3
    actor d = new dolphin(weight, age)
    
    // Testing formal args shadowing
    swim(13) | d
    eat(30) | d
    sleep(0) | d
    hello() | d
    die() | d

    return 0

