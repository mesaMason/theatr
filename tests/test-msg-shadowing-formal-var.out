inside dolphin.
dolphins local sum var =
13
inside swim().
dolphins hours =
0
inside eat().
eat()s formal weight =
30
inside sleep().
sleep()s local sum =
0
inside drop().
dolphins formal weight =
10
dolphins local sum =
13
